src/nvim/main.c

```diff
@@ -2057,93 +2057,66 @@ static void mainerr(int n, const char *str)
 }
 
 
-/*
- * print a message with three spaces prepended and '\n' appended.
- */
-static void main_msg(char *s)
-{
-  mch_msg("   ");
-  mch_msg(s);
-  mch_msg("\n");
-}
-
-/*
- * Print messages for "vim -h" or "vim --help" and exit.
- */
+/// Prints help message and exits; used for 'nvim -h' & 'nvim --help'
 static void usage(void)
 {
-  int i;
-  static char *(use[]) =
-  {
-    N_("[file ..]       edit specified file(s)"),
-    N_("-               read text from stdin"),
-    N_("-t tag          edit file where tag is defined"),
-    N_("-q [errorfile]  edit file with first error")
-  };
-
-  signal_stop();              /* kill us with CTRL-C here, if you like */
-
-  mch_msg(longVersion);
```

Comment:
I haven't seen it done anywhere else, but regardless I think it's just noise. I don't see how it's in any way helpful given that --version and the intro screen already include it.

On February 2, 2015 1:53:02 AM EST, "Justin M. Keyes" <notifications@github.com> wrote:
>>  static void usage(void)
>>  {
>> -  int i;
>> -  static char *(use[]) =
>> -  {
>> -    N_("[file ..]       edit specified file(s)"),
>> -    N_("-               read text from stdin"),
>> -    N_("-t tag          edit file where tag is defined"),
>> -    N_("-q [errorfile]  edit file with first error")
>> -  };
>> -
>> -  signal_stop();              /* kill us with CTRL-C here, if you
>like */
>> -
>> -  mch_msg(longVersion);
>
>Why not print `longVersion`? It's a pretty common tradition with unix
>tools.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/neovim/neovim/pull/1894/files#r23909221